traffic_ops/testing/api/v4/jobs_test.go

```diff
@@ -360,3 +372,843 @@ func GetTestInvalidationJobs(t *testing.T) {
+
+func CreateJobsMissingRegex(t *testing.T) {
+	if len(testData.InvalidationJobs) < 1 {
+		t.Fatal("Need at least one Invalidation Job to test creating an invalid Job")
+	}
+	//Missing Regex
+	//Future start date
+	dt := time.Now()
+	dt.Format("2019-10-12T07:20:50.52Z")
+	job := testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  dt.AddDate(0, 0, 1),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request := tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		TTL:             job.TTL,
+		StartTime:       job.StartTime,
+	}
+	resp, reqInf, err := TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected regex: cannot be blank, but no error found %v - Alert %v", resp, resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//Empty Regex
+	job.Regex = nil
+	request = tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		TTL:             job.TTL,
+		StartTime:       job.StartTime,
+	}
+	resp, _, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected regex: cannot be blank, but no error found %v - Alert %v", resp, resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateJobsMissingTtl(t *testing.T) {
+	if len(testData.InvalidationJobs) < 1 {
+		t.Fatal("Need at least one Invalidation Job to test creating an invalid Job")
+	}
+	//Missing TTL
+	//Future start date
+	dt := time.Now()
+	dt.Format("2019-10-12T07:20:50.52Z")
+	job := testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  dt.AddDate(0, 0, 1),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request := tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+	}
+	resp, reqInf, err := TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected ttl: cannot be blank., but no error found %v - Alert %v", resp, resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//Empty TTL
+	job.TTL = nil
+	request = tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		TTL:             job.TTL,
+		StartTime:       job.StartTime,
+	}
+	resp, _, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected ttl: cannot be blank., ttl: must be a number of hours, or a duration string e.g. '48h', but no error found %v - Alert %v", resp, resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+}
+
+func UpdateTestJobsInvalidDS(t *testing.T) {
```

Comment:
All update cases have been covered but I see one missing: https://github.comcast.com/cdn/cdn-tests/blob/master/trafficops/tests/jobs_test.go#L176 or did I misconstrues?

Reply:
I am not able to pass invalid Date to the date field. By default, it throws an error in the IDE. I feel there is no need to imply this validation

Reply:
Kindly provide your comments on the same.

Reply:
From what I understand, empty date implies `time.Time{}`, which resolves to `0001-01-01 00:00:00 +0000 UTC`. In other words that means a date set in past and we already have a test case for the past start date. So, no other change required.